test/unit/kubernetes-deploy/kubernetes_resource/daemon_set_test.rb

```diff
@@ -38,22 +40,78 @@ def test_deploy_passes_when_updated_available_does_match
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', uid: OWNER_UID, generation: 2, status: status)
+    pod_template = build_pod_template(filename: 'daemon_set_pod.yml')
+    node_template = build_node_template(filename: 'node.yml')
+    ds = build_synced_ds(ds_template: ds_template, pod_templates: pod_template, node_templates: node_template)
+    assert_predicate(ds, :deploy_succeeded?)
+  end
+
+  def test_deploy_passes_when_ready_pods_for_multiple_nodes
+    status = {
+      "desiredNumberScheduled": 3,
+      "updatedNumberScheduled": 3,
+      "numberReady": 2,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', uid: OWNER_UID, generation: 2, status: status)
+    pod_template = build_pod_template(filename: 'daemon_set_pod.yml')
+    node_template = build_node_template(filename: 'node.yml')
+    ds = build_synced_ds(ds_template: ds_template, pod_templates: pod_template, node_templates: node_template)
     assert_predicate(ds, :deploy_succeeded?)
   end
 
+  def test_deploy_fails_when_not_all_pods_updated
+    status = {
+      "desiredNumberScheduled": 3,
+      "updatedNumberScheduled": 2,
+      "numberReady": 2,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', uid: OWNER_UID, generation: 2, status: status)
+    pod_template = build_pod_template(filename: 'daemon_set_pod.yml')
+    node_template = build_node_template(filename: 'node.yml')
+    ds = build_synced_ds(ds_template: ds_template, pod_templates: pod_template, node_templates: node_template)
+    refute_predicate(ds, :deploy_succeeded?)
+  end
+
   private
 
-  def build_ds_template(status: {})
-    base_ds_maifest = YAML.load_stream(File.read(File.join(fixture_path('for_unit_tests'), 'daemon_set.yml'))).first
-    base_ds_maifest.deep_merge("status" => status)
+  def build_ds_template(filename:, status: {}, uid: nil, generation: nil)
+    base_ds_manifest = YAML.load_stream(File.read(File.join(fixture_path('for_unit_tests'), filename))).first
+    base_ds_manifest.deep_merge!("status" => status)
+    base_ds_manifest['metadata']['uid'] = uid if uid
+    base_ds_manifest['spec']['templateGeneration'] = generation if generation
+    base_ds_manifest
+  end
+
+  def build_pod_template(filename:)
+    base_pod_manifest = YAML.load(File.read(File.join(fixture_path('for_unit_tests'), filename)))
+    [base_pod_manifest].flatten
+  end
+
+  def build_node_template(filename:)
+    base_node_manifest = YAML.load(File.read(File.join(fixture_path('for_unit_tests'), filename)))
+    if base_node_manifest['kind'] == 'List'
+      base_node_manifest['items']
+    else
+      [base_node_manifest].flatten
+    end
   end
 
-  def build_synced_ds(template:)
-    ds = KubernetesDeploy::DaemonSet.new(namespace: "test", context: "nope", logger: logger, definition: template)
-    stub_kind_get("DaemonSet", items: [template])
-    stub_kind_get("Pod", items: [])
+  def build_synced_ds(ds_template:, pod_templates: nil, node_templates: nil)
```

Comment:
Suggestion: since the keywords are plural, make the default values `[]` and then remove the `|| []` below